tests/testthat/test-expect-length.R

```diff
@@ -1,12 +1,31 @@
 context("expect_length")
 
-test_that("fails if not a vector", {
-  expect_failure(expect_length(environment(), 1), "not a vector")
-})
-
 test_that("length computed correctly", {
   expect_success(expect_length(1, 1))
-  expect_failure(expect_length(1, 2))
+  expect_failure(expect_length(1, 2), "has length 1, not length 2.")
+  expect_success(expect_length(1:10, 10))
+  expect_success(expect_length(letters[1:5], 5))
+})
+
+test_that("uses S4 length method, if exists", {
+  # A has no length method defined
+  A <- setClass("ExpectLengthA", slots=c(x="numeric", y="numeric"))
+  # Default for S4 objects that don't inherit a length method: always length 1
+  expect_success(expect_length(A(x=1:5, y=3), 1))
+
+  # B does has a length method defined
```

Comment:
This mostly tests of the implementation of `length()`. I think you only need the test for `ExpectLengthB`

Reply:
Tests pruned.